modules/hostap/src/supp_api.c

```diff
@@ -1128,6 +1128,50 @@ int supplicant_pmksa_flush(const struct device *dev)
 	return ret;
 }
 
+int supplicant_11k_cfg(const struct device *dev, struct wifi_11k_params *params)
+{
+	const struct wifi_mgmt_ops *const wifi_mgmt_api = get_wifi_mgmt_api(dev);
+
+	if (!wifi_mgmt_api || !wifi_mgmt_api->cfg_11k) {
+		wpa_printf(MSG_ERROR, "cfg 11k not supported");
+		return -ENOTSUP;
+	}
+
+	return wifi_mgmt_api->cfg_11k(dev, params);
+}
+
+int supplicant_11k_neighbor_request(const struct device *dev, struct wifi_11k_params *params)
+{
+	int ssid_len = strlen(params->ssid);
+
+	if (params && ssid_len) {
+		if (ssid_len > WIFI_SSID_MAX_LEN) {
+			wpa_printf(MSG_ERROR, "%s: ssid too long %u",
+				   __func__, ssid_len);
+			return -1;
+		}
+
+		if (ssid_len) {
```

Comment:
This check seems redundant, the ssid_len cannot be 0 at this point.

Reply:
Removed this `ssid_len` check.